checker/jdk/nullness/src/java/util/concurrent/ConcurrentHashMap.java

```diff
@@ -0,0 +1,42 @@
+package java.util.concurrent;
+import org.checkerframework.dataflow.qual.Pure;
+import org.checkerframework.dataflow.qual.SideEffectFree;
+import java.util.*;
+import org.checkerframework.checker.nullness.qual.KeyFor;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class ConcurrentHashMap<K extends @NonNull Object, V extends @NonNull Object> extends AbstractMap<K, V>
+        implements ConcurrentMap<K, V>, java.io.Serializable {
+
+    private static final long serialVersionUID = 0;
+
+     public ConcurrentHashMap(int initialCapacity,float loadFactor, int concurrencyLevel) {throw new RuntimeException("skeleton method");}
+     public ConcurrentHashMap(int initialCapacity, float loadFactor) {throw new RuntimeException("skeleton method");}
+     public ConcurrentHashMap(int initialCapacity) {throw new RuntimeException("skeleton method");}
+     public ConcurrentHashMap() {throw new RuntimeException("skeleton method");}
+     @Pure public boolean isEmpty() {throw new RuntimeException("skeleton method");}
+     @Pure public int size() {throw new RuntimeException("skeleton method");}
+     @Pure public @Nullable V get(@NonNull Object key){throw new RuntimeException("skeleton method");}
+     @Pure public boolean containsKey(@NonNull Object key) {throw new RuntimeException("skeleton method");}
+     @Pure public boolean containsValue(@NonNull Object value) {throw new RuntimeException("skeleton method");}
+     @Pure public boolean contains(@NonNull Object value) {throw new RuntimeException("skeleton method");}
+     public @Nullable V put(@NonNull K key,@NonNull V value) {throw new RuntimeException("skeleton method");}

```

Comment:
Yes it didn't work that way.

Consider this example:

import java.util.concurrent.ConcurrentHashMap;
import org.checkerframework.checker.nullness.qual.Nullable;

class Issue1142 {

    void foo() {
        //:: error: (type.argument.type.incompatible)
        ConcurrentHashMap<Integer, @Nullable Integer> chm1 = new ConcurrentHashMap<>();

        //:: error: (argument.type.incompatible)
        chm1.put(1, null);
    }
}

It just produced the 1st error ignoring the 2nd one completely. Then after specifying @NonNull annotation before the K & V key it produced both the errors simultaneously. So i opted to use it this way.